src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -874,6 +874,7 @@ public PreprocessingSymbolInfo GetPreprocessingSymbolInfo(SyntaxNode nameSyntax)
         internal abstract void ComputeDeclarationsInNode(SyntaxNode node, ISymbol associatedSymbol, bool getSymbol, ArrayBuilder<DeclarationInfo> builder, CancellationToken cancellationToken, int? levelsToCompute = null);
 
         internal virtual Func<SyntaxNode, bool>? GetSyntaxNodesToAnalyzeFilter(SyntaxNode declaredNode, ISymbol declaredSymbol) => null;
+        internal virtual bool ShouldSkipSyntaxNodeAnalysis(SyntaxNode declaredNode, ISymbol declaredSymbol) => false;
```

Comment:
>ShouldSkipSyntaxNodeAnalysis [](http://example.com/codeflow?start=30&length=28)

It is unfortunate that you were not able to clearly separate the two filters. I was hoping that we would be able to rename the previous helper to something like "DescendIntoChildrenSyntaxNodesToAnalyzeFilter" and have its implementation follow the new purpose. However, I can see that that might require some non-trivial changes in the implementation of the helper and it is fine to avoid the additional risk and complexity at this point. However, I think it would be good to add some comment clarifying the distinction between the helpers for the benefit of implementers. Otherwise, an implementer could be confused where exactly a node should be filtered and not filtered out. For example, right now one might ask a question: "Why do we need the ```ShouldSkipSyntaxNodeAnalysis``` when ```GetSyntaxNodesToAnalyzeFilter``` could filter that node out. It would be good to have this question answered in a doc comment.